src/msw/dib.cpp

```diff
@@ -656,10 +762,19 @@ bool wxDIB::Create(const wxImage& image, PixelFormat pf)
         {
             for ( int x = 0; x < w; x++ )
             {
-                *dst++ = src[2];
-                *dst++ = src[1];
-                *dst++ = src[0];
-                src += 3;
+                if ( bpp != 1 )
```

Comment:
I haven't measured it, but I think it could be better to move this test out of the loop to allow compiler to handle the loop more efficiently (e.g. auto-vectorize it).